### PR TITLE
Switch to using subprocess::shell

### PR DESCRIPTION
**Pull request by jonathandturner**
Switch to using the shell for subprocess to enable more natural shelling out.  This should allow calling out to .cmd scripts, .sh scripts, and so on.

### CODE REVIEW

File: src/commands/classified/external.rs [hunk 1]
```diff
@@ -254,7 +254,7 @@ async fn spawn(
         process = process.arg(&arg);
     }
 
-    process = process.cwd(path);
```
Comment (thegedge) [inline src/commands/classified/external.rs]: This sets the current working directory for the process. I think you want to tweak line 251 to be `Exec::shell`?

Comment (jonathandturner) [reply src/commands/classified/external.rs]: Good catch

Comment (jonathandturner) [reply src/commands/classified/external.rs]: That's what I get for doing the edit when I'm not paying attention :)

File: src/commands/classified/external.rs [hunk 2]
```diff
@@ -209,8 +209,12 @@ async fn run_with_stdin(
 
             let process_args = args.iter().map(|arg| {
                 let arg = expand_tilde(arg.deref(), || home_dir.as_ref());
-                if let Some(unquoted) = remove_quotes(&arg) {
-                    unquoted.to_string()
+                if argument_contains_whitespace(&arg) && argument_is_quoted(&arg) {
+                    if let Some(unquoted) = remove_quotes(&arg) {
+                        format!("'{}'", unquoted)
```
Comment (jonathandturner) [inline src/commands/classified/external.rs]: I realized this won't work if the pass in a string with a single quote like "Sam's PR"

File: src/commands/classified/external.rs [hunk 3]
```diff
@@ -248,11 +252,18 @@ async fn spawn(
...
-    for arg in args {
-        process = process.arg(&arg);
-    }
+    let mut process = Exec::shell(&cmd_with_args);
+
+    let paths = shell_os_paths();
+
+    let paths_joined = match std::env::join_paths(paths.iter()) {
+        Ok(all) => all,
+        Err(_) => panic!("Internal error: Couldn't join paths for PATH var."),
```
Comment (jonathandturner) [inline src/commands/classified/external.rs]: We should probably just return a ShellError instead of panicking

**Pull request merged**